Algorithm.CSharp/FuturesExpiredContractRegression.cs

```diff
@@ -0,0 +1,99 @@
+﻿/* 
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm to test if expired futures contract chains are making their
+    /// way into the timeslices being delivered to OnData()
+    /// </summary>
+    public class FuturesExpiredContractRegression : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        /// <summary>
+        /// Initializes the algorithm state.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 1);
+            SetEndDate(2013, 12, 23);
+            SetCash(1000000);
+
+            // Subscribe to futures ES
+            var future = AddFuture(Futures.Indices.SP500EMini, Resolution.Minute, Market.CME, false);
+            future.SetFilter(TimeSpan.FromDays(0), TimeSpan.FromDays(90));
+        }
+
+        public override void OnData(Slice data)
+        {
+            foreach (var chain in data.FutureChains)
+            {
+                foreach (var contract in chain.Value.OrderBy(x => x.Expiry))
+                {
+                    if (contract.Expiry.Date < Time.Date)
+                    {
+                        throw new Exception($"Received expired contract {contract} expired: {contract.Expiry} current time: {Time}");
+                    }
+                }
+            }
+        }
```

Comment:
Cool! We should have a secondary assert to make sure OnData was actually called with data, we usually do this opening a position in the related security but it could also be through some assert in OnAlgorithmEnded I think, the same comment about the option regression algorithm